### PR TITLE
arm64: dts: qcom: msm8916-motorola-surnia: Enable modem, sound and blsp_i2c1

### PR DESCRIPTION
**Pull request by linuxgorl**
osprey and surnia share the same common dtsi for sound in downstream kernel

### CODE REVIEW

**Comment by Minecrell** [comment]
Hi @linuxgorl, CI is failing here, please squash your commits and take a look at the contributing guidelines: https://github.com/msm8916-mainline/linux/blob/master/CONTRIBUTING.md

**Comment by linuxgorl** [comment]
Should be good now. I've asked @unrznbl to test this patch on his surnia variants

**Comment by Minecrell** [comment]
Hi @linuxgorl. Unfortunately upstream Linux requires commits to be signed-off with a real name (`FirstName LastName <e@mail>`). This isn't a requirement I came up with and therefore I can't change it. Not following it would mean that your commit cannot be submitted for inclusion upstream (on the mailing lists). Please let me know here or on Matrix if you have any concerns about this.

**Comment by craigcomstock** [comment]
> Should be good now. I've asked @unrznbl to test this patch on his surnia variants

On one variant I don't think things are working with the firmware provided by postmarketos build. I get this over and over in dmesg:

```
[   60.106028] qcom-wcnss-pil a204000.remoteproc: unexpected response to sysmon event
[   60.166393] qcom-q6v5-mss 4080000.remoteproc: MBA booted without debug policy, loading mpss
[   60.804913] remoteproc remoteproc0: remote processor 4080000.remoteproc is now up
[   60.829523] qcom-q6v5-mss 4080000.remoteproc: fatal error received: fs_device_efs_rmts.c:154:[8 ,917504, 1] EFS: rmts_get_buffer api failed
[   60.829609] remoteproc remoteproc0: crash detected in 4080000.remoteproc: type fatal error
[   60.829719] remoteproc remoteproc0: handling crash #15 in 4080000.remoteproc
[   60.829768] remoteproc remoteproc0: recovering 4080000.remoteproc
[   62.630222] qcom-wcnss-pil a204000.remoteproc: unexpected response to sysmon event
[   62.630243] remoteproc remoteproc0: stopped remote processor 4080000.remoteproc
[   62.631100] qcom-wcnss-pil a204000.remoteproc: unexpected response to sysmon event
[   62.690406] qcom-q6v5-mss 4080000.remoteproc: MBA booted without debug policy, loading mpss
[   63.325788] remoteproc remoteproc0: remote processor 4080000.remoteproc is now up
[   63.356073] qcom-q6v5-mss 4080000.remoteproc: fatal error received: fs_device_efs_rmts.c:154:[8 ,917504, 1] EFS: rmts_get_buffer api failed
[   63.356107] remoteproc remoteproc0: crash detected in 4080000.remoteproc: type fatal error
[   63.356138] remoteproc remoteproc0: handling crash #16 in 4080000.remoteproc
[   63.356149] remoteproc remoteproc0: recovering 4080000.remoteproc
[   65.157728] qcom-wcnss-pil a204000.remoteproc: unexpected response to sysmon event
[   65.157748] remoteproc remoteproc0: stopped remote processor 4080000.remoteproc
```
The interesting bits about this device from `fastboot getvar all` are

```
...
(bootloader) hwrev: 0x8300
(bootloader) radio: 0x1
...
(bootloader) version: 0.5
(bootloader) version-bootloader: moto-msm8916-80.10
...
(bootloader) version-baseband[0]: M8916_2020602.05.03.21.07R SURNIA_TMO_
(bootloader) version-baseband[1]: CUST
```

I can try to pull different firmware files from the device itself and see if that helps. That was necessary for me on harpia.

**Comment by craigcomstock** [comment]
I tried the modem.* and mba.mbn from my "modem" partition and got these errors:

```
[   24.404616] remoteproc remoteproc0: powering up 4080000.remoteproc
[   24.408018] remoteproc remoteproc0: Booting fw image mba.mbn, size 230208
[   24.409316] qcom-wcnss-pil a204000.remoteproc: unexpected response to sysmon event
[   24.474053] qcom-q6v5-mss 4080000.remoteproc: MBA booted without debug policy, loading mpss
[   25.272947] remoteproc remoteproc0: remote processor 4080000.remoteproc is now up
[   25.361625] qcom-q6v5-mss 4080000.remoteproc: fatal error received: fs_device_efs_rmts.c:154:[8 ,917504, 1] EFS: rmts_get_buffer api failed
[   25.361662] remoteproc remoteproc0: crash detected in 4080000.remoteproc: type fatal error
[   25.361787] remoteproc remoteproc0: handling crash #1 in 4080000.remoteproc
[   25.361802] remoteproc remoteproc0: recovering 4080000.remoteproc
```

Will try the original packaged firmware to compare. :)

**Comment by craigcomstock** [comment]
Yes. Same set of errors and I check several of the firmware files and my local partition files are different than those from pmos packaging. So maybe not a firmware issue.

```
surnia:/lib/firmware# md5sum  dist.pmos.modem/mba.mbn
b7f76387fb08e3d70079045c7459d825  dist.pmos.modem/mba.mbn
surnia:/lib/firmware# md5sum local.modem/mba.mbn
28d0384b1dff75ddf469f62766f77da0  local.modem/mba.mbn
```

I hacked pmaports to build the kernel though, so will try envkernel instead now and compare.

**Comment by craigcomstock** [comment]
Not sure if I am making the kernel correctly. Using bits and pieces from here https://wiki.postmarketos.org/wiki/Mainlining_Guide#envkernel.sh but for example, no zImage but rather an Image.gz, same thing? How can I confirm I am running the built kernel?

**Comment by linuxgorl** [comment]
do `rc-service rmtfs stop`
and `sudo rmtfs -P -r -s -v`

**Comment by linuxgorl** [comment]
it should complain about surnia_emea_4

**Comment by craigcomstock** [comment]
> it should complain about surnia_emea_4

most interesting message seems to be

`[RMTFS] open [RMTFS storage] request for unknown partition '/boot/modem_fsg/surnia_tmo_1', rejecting`

**Comment by linuxgorl** [comment]
okay, mount modem_fsg and you should see a gzipped surnia_tmo_1 image.

rmtfs can't read those images atm, that's why we do an ugly hack:

gunzip the image, patch rmtfs like [this](https://github.com/linuxgorl/rmtfs/commit/e3578a6d97a959cfd3a38f201c10c4033d4a803e) and move the image to /dev/disk/by-partlabel/

**Comment by unrznbl** [comment]
@linuxgorl after making the change you suggested and copying surnia_tmo_1.img to various locations, it seems `/boot/modem_fsg/tmo_1` is what made things work I was able to bring up the modem with `ofonod -d -n` and `ofonoctl`:

```
surnia:/etc/init.d# ofonoctl
Modem    Status     SIM
-------  ---------  -------
gobi_0   Unpowered  Unknown
surnia:~$ ofonoctl  poweron
Powered on gobi_0
surnia:~$ ofonoctl  online
Brought gobi_0 online
surnia:~$ ofonoctl
Modem    Status                     SIM
-------  -------------------------  -------
gobi_0   Registered to H2O (80.0%)  Unknown
```

I am able to make a voice call but no audio just yet (might be my shelli-ui business needs fixing here). So I'd say generally I give this a thumbs up. :)

**Comment by unrznbl** [comment]
Oh. I was able to make and receive phone calls and SMS just fine. Just no audio for me yet, but like I said that's almost certainly a shelli-ui issue. :) I'll be working on debugging and fixing that in pmaports or shelli itself.

**Comment by unrznbl** [comment]
Wasn't sure if rmnet was supposed to work via bam_dmux but didn't work for me:

```
surnia:~# dhcpcd rmnet0
rmnet0: unsupported interface family 0x207
rmnet0: does not have hardware address
picked interface usb0 to generate a DUID
DUID 00:01:00:01:28:09:15:e4:d2:37:fd:21:f2:a0
rmnet0: IAID 00:00:00:04
rmnet0: waiting for 3rd party to configure IP address
timed out
dhcpcd exited
```

In dmesg I get `[ 1222.588852] bam-dmux wwan: Channel already open: 0` for 0 through 8 FWIW.

I tried setting the APN as mentioned in https://gist.github.com/Minecrell/4cc2bfb9fcae18e294386b0a213907d1 but that didn't seem to change the result.

I can try modemmanager another time.

**Comment by linuxgorl** [comment]
rmnet works with bam-dmux and rawip now

**Comment by Minecrell** [comment]
@linuxgorl I think it would be good to find a solution for this rmtfs stuff before we merge this. It might help if you try to ask Bjorn Andersson (bamse) for advice, either on IRC (`##linux-msm`) or with an issue at https://github.com/andersson/rmtfs.

File: arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts [hunk 1]
```diff
@@ -45,6 +73,10 @@
 	};
 };
 
+&blsp_i2c1 {
+	status = "okay";
+};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts:78]: This seems unused?

File: arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts [hunk 2]
```diff
@@ -166,6 +202,40 @@
...
+		cpu {
+			sound-dai = <&lpass MI2S_TERTIARY>;
+		};
+		codec {
+			sound-dai = <&lpass_codec 1>, <&wcd_codec 1>;
+		};
+	};
+};
+
+
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts:238]:
```suggestion
```

File: arch/arm64/boot/dts/qcom/msm8916-motorola-surnia-modem.dts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+			sound-dai = <&q6afedai TERTIARY_MI2S_TX>;
+		};
+		platform {
+			sound-dai = <&q6routing>;
+		};
+		codec {
+			sound-dai = <&lpass_codec 1>, <&wcd_codec 1>;
+		};
+	};
+};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-motorola-surnia-modem.dts:34]: Please keep this in an extra commit (see other devices for an example for the modem commit)

File: arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts (hunk 1 above)
Comment (linuxgorl) [reply arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts:78]: It's for the backlight, I plan to rewrite the driver, should I remove it? It works fine with it

Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts:78]: You should add it when you add the backlight device to it. So yes, remove it for now.

**Comment by Minecrell** [comment]
@linuxgorl Thanks for making the changes. Can you fix the CI failures? It complains about using spaces instead of tabs for indentation.

**Comment by linuxgorl** [comment]
> @linuxgorl Thanks for making the changes. Can you fix the CI failures? It complains about using spaces instead of tabs for indentation.

Should be good now. Once this gets merged I will write the instructions on the wiki

**Review by Minecrell — APPROVED** [review]

**Pull request merged**